dora/core/server/master/src/main/java/alluxio/master/scheduler/Scheduler.java

```diff
@@ -139,24 +139,18 @@ public boolean enqueueTaskForWorker(@Nullable WorkerInfo workerInfo, Task task,
         return false;
       }
       if (kickStartTask) {
+        // track running tasks of a job
+        ConcurrentHashSet<Task<?>> tasks = mJobToRunningTasks.computeIfAbsent(task.getJob(),
+                j -> new ConcurrentHashSet<>());
+        tasks.add(task);
         mScheduler.getWorkingExecutor().submit(() -> {
```

Comment:
submit will also throw exception, needs to be caught and do remove. We can leverage the section 165-168 to removeTaskFromWorkerQ, and let's guard the mJobToRunningTasks and mWorkerToTaskQ with a reentrantlock